cmd/util/ledger/migrations/staged_contracts_migration.go

```diff
@@ -144,10 +171,17 @@ func (m *StagedContractsMigration) MigrateAccount(
 		return payloads, nil
 	}
 
+	elaborations := map[common.Location]*sema.Elaboration{}
+
 	config := util.RuntimeInterfaceConfig{
 		GetContractCodeFunc: func(location runtime.Location) ([]byte, error) {
 			return m.contractsByLocation[location], nil
 		},
+		GetOrLoadProgramListener: func(location runtime.Location, program *interpreter.Program, err error) {
+			if err == nil {
+				elaborations[location] = program.Elaboration
+			}
+		},
```

Comment:
Is this safe? or is there a better way to collect the list of elaborations?

Reply:
This is probably fine